lib/js/frame.js

```diff
@@ -0,0 +1,72 @@
+var Frame = function(game) {
+  this.game = game;
+  this.total = 0;
+  this.rolls = 0;
+  this.bonusRolls = 0;
+};
+
+Frame.prototype.accepts = function(value) {
+  var pinsRemaining = 10 - (this.total % 10);
+  if (value > pinsRemaining) return false;
+  if (value < 0) return false;
+  if (!Number.isInteger(value)) return false;
+  return true;
+};
+
+Frame.prototype.add = function(value) {
+  this.total += value;
+};
+
+Frame.prototype.roll = function(value) {
+  this._expendBonuses(value);
+  this.add(value);
+  this.rolls ++;
+  if (this.total === 10) this._setBonus();
+};
+
+Frame.prototype.isComplete = function() {
+  if (this._isFinal()) return this._tenthFrameComplete();
+  else return this._normalComplete();
+};
+
+Frame.prototype.hasBonus = function() {
+  return this.bonusRolls > 0;
+};
+
+Frame.prototype.expendBonus = function(value) {
+  if (this.hasBonus()) {
+    this._reduceBonus();
+    if (!this._isFinal()) this.add(value);
+  }
+};
+
+Frame.prototype._expendBonuses = function(value) {
+  this.game.frames.forEach(function(frame) {
+    frame.expendBonus(value);
+  });
+};
+
+Frame.prototype._setBonus = function() {
+  if (this.rolls == 1) this.bonusRolls += 2; // strike
+  else this.bonusRolls += 1; // spare
```

Comment:
These two comments suggest that there's information you're neglecting to put directly into your code. Why not make them more explicit, using a variable, or a constant (a variable that never changes)?